src/main.rs

```diff
@@ -24,25 +31,41 @@ fn time_to_mine(block_height: u64) -> Duration {
     // when the lazy macro is expanded
     // if a value has a static lifetime then it means that value lives as long as the program lives
     let rpc_client: &Client = &*RPC_CLIENT;
-    rpc_client.get_block_hash(234);
-    todo!()
+    let block: Block = get_block(block_height, rpc_client).unwrap();
+    println!("block={:?}", block);
+    let params = Params::new(Network::Bitcoin);
+    println!("params={:?}", params);
+
+    let difficulty = block.header.difficulty(params);
+    // finding it difficult to get the hash_rate from the crate
+    let hash_rate = 1.0;
+
+    let time = time_to_mine_block(difficulty as f64, hash_rate);
+
+    // i was supposed to return the time variable returned from the helper function
+    // so, i decided to use the time in the block header for the main time.
+    // I will appreciate if I can get a pointer to clarify this solution.
+    Duration::new(block.header.time as i64, 1).unwrap()
```

Comment:
This will just return the block timestamp as a Duration. 

This doesn't indicate how long it took to mine the block.

Reply:
Is there a way to get the hash rate of the block? I have a comment explaining it there in the code. @sb1752

Reply:
What you're looking for is the time to mine the block. Think about how you might get that number. Assume that mining starts immediately after the last block timestamp.

Reply:
Hi @sb1752, kindly review